internal/controller/volsync/vshandler.go

```diff
@@ -1441,7 +1444,9 @@ func (v *VSHandler) getVolumeSnapshotClassFromPVCStorageClass(storageClass *stor
 	var matchedVolumeSnapshotClassName string
 
 	for _, volumeSnapshotClass := range volumeSnapshotClasses {
-		if volumeSnapshotClass.Driver == storageClass.Provisioner {
+		sIDFromSnapClass := volumeSnapshotClass.GetLabels()[StorageIDLabel]
```

Comment:
For the upgrade, I'll assume that the `volumeSnapshotClass` labels will be updated automatically with the `StorageIDLabel`. Otherwise, this will fail given that the storageClass already have a `StorageIDLabel`.

Reply:
The function `isAsyncVSClassPeer` that forms peers will ignore this SC, so peers will be empty and fall back to older scheme of comparing VSC to SC to PVC etc.

At least that is the upgrade scenario I am looking at, and based on the code that should hold, till the managed clusters are also upgraded and labelled.